server/routes/notification.js

```diff
@@ -0,0 +1,17 @@
+const express = require("express");
+const router = express.Router();
+const protect = require("../middleware/auth");
```

Comment:
This isn't the most important concern for the scope of this project so I'll mark this as `optional`, but it's important that we are aware:

As is, this middleware only confirms that _a_ user is logged in, however, it doesn't guarantee that the user logged in is the same user as the one sent in the request body, under the `to` (or `from`) field. This creates a potential security concern where any user can forge a request to access other users' notifications. 

For a quick fix, we can add an additional middleware to the routes below, as necessary, that confirms `req.body.to` and the `req.user.id` are the same before proceeding (e.g. to `getNotification` and `getAllNotifications`). Similarly, we can apply the same check to `body.req.from` to `createNotification`. 

For `markNotification`, we can use `findOneAndUpdate` and look for a notification with the given notification id and given user in the `to` field.